src/mock.ts

```diff
@@ -91,57 +101,68 @@ function addMockFunctionsToSchema({ schema, mocks = {}, preserveResolvers = fals
     }
     return customMock;
   }
 
-  function assignResolveType(type) {
+  function getResolveType(namedFieldType: GraphQLNamedType) {
+    if ( (namedFieldType instanceof GraphQLInterfaceType) ||
+         (namedFieldType instanceof GraphQLUnionType) ) {
+        return namedFieldType.resolveType;
+    } else {
+        return undefined;
+    }
+  }
+
+  function assignResolveType(type: GraphQLType) {
     const fieldType = getNullableType(type);
     const namedFieldType = getNamedType(fieldType);
 
-    const oldResolveType = namedFieldType.resolveType;
+    const oldResolveType = getResolveType(namedFieldType);
     if (preserveResolvers && oldResolveType && oldResolveType.length) {
       return;
     }
 
     if (namedFieldType instanceof GraphQLUnionType ||
-        namedFieldType instanceof GraphQLInterfaceType
-    ) {
+        namedFieldType instanceof GraphQLInterfaceType) {
       // the default `resolveType` always returns null. We add a fallback
       // resolution that works with how unions and interface are mocked
-      namedFieldType.resolveType = (data, context, info) =>
-        info.schema.getType(data.typename);
+      namedFieldType.resolveType = (data: any, context: any, info?: GraphQLResolveInfo) => {
+        if ( undefined === info ) {
+            throw new Error(`no info provided for resolveType`);
+        }
+        return info.schema.getType(data.typename) as GraphQLObjectType;
+      };
     }
   }
 
-  const mockType = function mockType(type, typeName, fieldName) {
+  const mockType = function mockType(type: GraphQLType, typeName?: string, fieldName?: string): IResolveFn {
     // order of precendence for mocking:
     // 1. if the object passed in already has fieldName, just use that
     // --> if it's a function, that becomes your resolver
     // --> if it's a value, the mock resolver will return that
     // 2. if the nullableType is a list, recurse
     // 2. if there's a mock defined for this typeName, that will be used
     // 3. if there's no mock defined, use the default mocks for this type
-    return (...args) => {
-      const [root, queryArgs, context, info] = args;
-
+    return (root: any, args: { [key: string]: any }, context: any, info: GraphQLResolveInfo): any => {
       // nullability doesn't matter for the purpose of mocking.
       const fieldType = getNullableType(type);
       const namedFieldType = getNamedType(fieldType);
 
       if (root && typeof root[fieldName] !== 'undefined') {
-        let result;
+        let result: any;
+
         // if we're here, the field is already defined
         if (typeof root[fieldName] === 'function') {
-          result = root[fieldName](...args);
+          result = root[fieldName](root, args, context, info);
           if (result instanceof MockList) {
-            result = result.mock(...args, fieldType, mockType);
+            result = result.mock(root, args, context, info, <GraphQLList> fieldType, mockType);
```

Comment:
Ok, I figured as much. Is there a reason for that convention? The `<>` style seems a lot more in line with most programming languages.